http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AzurePubsubHandler.scala

```diff
@@ -923,69 +921,70 @@ class AzurePubsubHandlerInterp[F[_]: Parallel](
       ctx <- ev.ask
       auth <- samDAO.getLeoAuthToken
 
-      wsmResourceId <- F.fromOption(
-        msg.wsmResourceId,
-        DiskDeletionError(
-          msg.diskId,
-          msg.workspaceId,
-          s"No associated WsmResourceId found for Azure disk"
-        )
-      )
-
-      _ <- deleteDiskResource(Right(wsmResourceId), msg.workspaceId, auth)
-
-      deleteJobId = getWsmJobId("delete-disk", wsmResourceId)
-      getDeleteJobResult = wsmDao.getDeleteDiskJobResult(
-        GetJobResultRequest(msg.workspaceId, deleteJobId),
-        auth
-      )
-
-      taskToRun = for {
-        // We need to wait until WSM deletion job to be done to update the database
-        resp <- streamFUntilDone(
-          getDeleteJobResult,
-          config.deleteDiskPollConfig.maxAttempts,
-          config.deleteDiskPollConfig.interval
-        ).compile.lastOrError
+      // TODO: Check if VM is deleted in WSM here instead of passing through id from front leo
+      _ <- msg.wsmResourceId match {
+        case Some(wsmResourceId) =>
+          for {
+            jobId <- deleteDiskResource(wsmResourceId, msg.workspaceId, auth)
+            getDeleteJobResult = wsmDao.getDeleteDiskJobResult(
+              GetJobResultRequest(msg.workspaceId, jobId),
+              auth
+            )
 
-        _ <- resp.jobReport.status match {
-          case WsmJobStatus.Succeeded =>
-            for {
-              _ <- logger.info(ctx.loggingCtx)(s"disk ${msg.diskId.value} is deleted successfully")
-              _ <- dbRef.inTransaction(persistentDiskQuery.delete(msg.diskId, ctx.now))
+            // We need to wait until WSM deletion job to be done to update the database
+            taskToRun = for {
+              resp <- streamFUntilDone(
+                getDeleteJobResult,
+                config.deleteDiskPollConfig.maxAttempts,
+                config.deleteDiskPollConfig.interval
+              ).compile.lastOrError
+
+              _ <- resp.jobReport.status match {
+                case WsmJobStatus.Succeeded =>
+                  for {
+                    _ <- logger.info(ctx.loggingCtx)(s"disk ${msg.diskId.value} is deleted successfully")
+                    _ <- dbRef.inTransaction(persistentDiskQuery.delete(msg.diskId, ctx.now))
+                  } yield ()
+                case WsmJobStatus.Failed =>
+                  F.raiseError[Unit](
+                    DiskDeletionError(
+                      msg.diskId,
+                      msg.workspaceId,
+                      s"WSM delete disk job failed due to ${resp.errorReport.map(_.message).getOrElse("unknown")}"
+                    )
+                  )
+                case WsmJobStatus.Running =>
```

Comment:
this is interesting ... if we are "running", we consider the job status to be failed? 🤔

Reply:
this is the same as it was before...but it should be this error instead

`
DiskDeletionError(msg.diskId, msg.workspaceId, s"WSM delete disk job failed due to ${resp.errorReport.map(_.message).getOrElse("unknown")}")`

ie the job timed out according to our maxAttempts